### PR TITLE
Facebook Non-standard Login/Like/Share Overlay 

### PR DESCRIPTION
**Pull request by maxxcrawford**
Fixing part of #258. 

This pull-request adds an overlay to all Facebook-specific on-page assets. It targets them in a few ways: 

+ Any element that has the a class of `FacebookConnectButton`
+ Any `href` attribute that contains `facebook`
+ Any `data-oauthserver` attribute that contains `facebook`
+ Any element with a `title` of `share on facebook`
+ Any element with an `aria-label` of `Facebook`

## Known Issues

+ The current method of `HREF` containing `facebook` is too broad. If you search for Facebook on Google, you can see how this doesn't match correctly. 
+ This doesn't address standard implementations of FB login. This will be addressed in a seperate pull-request.  

## Testing

Recreation Steps: 
+ [Run this version](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Getting_started_with_web-ext#Using_web-ext) of the web extension locally
+ Test one of the test sites

Expected results: 
+ All Facebook subresources will be blocked. (Each attempted resource call will get a console.log) 
+ All on-page Facebook elements get an overlay/slash through it. 

Example test pages: 
+ [Buzzfeed](https://www.buzzfeed.com/benhenry/avengers-endgame-mark-ruffalo-streaming-thor )
+ [Imgur](https://imgur.com/signin)
+ [Amazon](https://www.amazon.com/AmazonBasics-Mini-DisplayPort-Thunderbolt-Adapter/dp/B00NH13K8S)
+ [Pinterest](https://www.pinterest.com/pin/326862885453847891/)
+ [IMDb](https://www.imdb.com/title/tt0448115/) 

## Example Screenshots

image

image

image

image

image

### CODE REVIEW

**Comment by maxxcrawford** [comment]
Revised screenshots to latest commit.

File: src/content_script.js [hunk 1]
```diff
@@ -1,6 +1,111 @@
 "use strict";
 
+function itemWidthCheck(target){
+
+  const itemHeight = target.offsetHeight;
+  const itemWidth = target.offsetWidth;
+  let iconClassArr = ["fbc-overlay"];
```
Comment (groovecoder) [inline src/content_script.js]: Go ahead and use `const` here - you can still call `.push` on it, and it protects us from accidental re-assignment later.

File: src/content_script.js [hunk 2]
```diff
@@ -1,6 +1,111 @@
...
+  }
+  return iconClassArr;
+}
+
+function detectFacebookOnPage () {
+
+  // TODO: Loop through array of class names to test for
+  // Test for any element with facebook connect in class name (Example: Pinterest)
+
+  const loginButtonTitle = document.querySelectorAll("[title*='Facebook']");
```
Comment (groovecoder) [inline src/content_script.js]: Let's consolidate all these into a single ...
```js
for ( let fbElementSelector of FACEBOOK_ELEMENT_SELECTORS) {
  for (let fbElement of document.querySelectorAll(fbElementSelector) ) {
    if ( !item.classList.contains("fbc-overlay") ) {
      item.classList.add(...itemWidthCheck(item));
    }
  }
}
```

File: src/content_script.js (hunk 1 above)
Comment (maxxcrawford) [reply src/content_script.js]: Revised!

File: src/content_script.js (hunk 2 above)
Comment (maxxcrawford) [reply src/content_script.js]: Done.

File: src/content_script.js [hunk 3]
```diff
@@ -1,108 +1,48 @@
...
+  "[title*='Facebook']",
+  "[title='Log in with Facebook']",
+  "[class*='fb-login']",
+  "[class*='FacebookConnectButton']",
+  "[href*='facebook']",
+  "[data-bfa-network*='facebook']",
+  "[data-oauthserver*='facebook']",
+  "[aria-label*='Facebook']",
+  "[data-destination*='facebook']"
+];
```
Comment (groovecoder) [inline src/content_script.js]: We'll file follow-up issues to add to this list as needed.

**Comment by groovecoder** [comment]
Merging with the console messages for now. We'll remove by the time we merge `2.0` into `master`.

**Pull request merged**